File: code/simian/simian-master-v1.6/SimianJS/Tests/test.Q.js
//#!/usr/bin/env js --enable-avx
const load = masala.io.load, print = masala.io.print, random = Math.random, now = masala.time.now;
//const load = masala.io.load, print = masala.io.print, random = masala.random.uniform, now = masala.time.now;

load("./eventQ.js");
const COUNT = 10000000;

var myList = [];
var Q = new eventQ(myList);

print("Pushing:");
for (var i=0; i<COUNT; i++) Q.push({"time" : random()});

print("Checking Queue Pop Order:");
var prvMin = -1;
var topItem;
while (myList.length > 0) {
    topItem = Q.pop();
    if (prvMin > topItem.time) print("Out of order");
    prvMin = topItem.time;
}

print("Pushing Time Check:");
var start = now();
for (var i=0; i<COUNT; i++) Q.push({"time" : random()});
var finish = now() - start;
print("One Push took: " + finish/COUNT + " ms")

print("Poping Time Check:");
var topItem;
var start = now();
while (myList.length > 0) topItem = Q.pop();
var finish = now() - start;
print("One Pop took: " + finish/COUNT + " ms")

File: code/simian/simian-master-v1.6/SimianJS/Tests/test.Q.lua
#!/usr/bin/env luajit
local Q = require("eventQ");
local COUNT = 10000000;

local myList = {};

print("Pushing:");
for i=1,COUNT do
    Q.push(myList, {time = math.random()});
end

print("Checking Queue Pop Order:");
local prvMin = -1;
local topItem;
while (#myList > 1) do
    topItem = Q.pop(myList);
    if (prvMin > topItem.time) then print("Out of order"); end
    prvMin = topItem.time;
end

print("Pushing Time Check:");
local start = os.clock();
for i=1,COUNT do
    Q.push(myList, {time = math.random()});
end
 finish = os.clock() - start;
print("One Push took: " .. (1000*finish/COUNT) .. " ms")

print("Poping Time Check:");
local topItem;
local start = os.clock();
while (#myList > 1) do
    topItem = Q.pop(myList);
end
local finish = os.clock() - start;
print("One Pop took: " .. (1000*finish/COUNT) .. " ms")
